File: packages/shared-types/src/languages.ts
import { z } from 'zod';

export const LanguageCodeSchema = z.enum([
  'en',
  'hi',
  'kn',
  'ta',
  'te',
  'mr',
  'bn',
  'ml',
  'gu',
  'pa',
]);
export type LanguageCode = z.infer<typeof LanguageCodeSchema>;

export const ScriptSchema = z.enum([
  'latin',
  'devanagari',
  'kannada',
  'tamil',
  'telugu',
  'malayalam',
  'gurmukhi',
  'gujarati',
  'bengali',
]);
export type Script = z.infer<typeof ScriptSchema>;

export const LanguageConfigSchema = z.object({
  code: LanguageCodeSchema,
  name: z.string().min(1),
  proficiency: z.enum(['primary', 'secondary']),
  script: ScriptSchema.optional(),
});
export type LanguageConfig = z.infer<typeof LanguageConfigSchema>;

File: packages/python-shared/veda_shared/schemas/languages.py
from __future__ import annotations

from enum import Enum
from typing import Literal

from pydantic import BaseModel, ConfigDict


class LanguageCode(str, Enum):
    EN = "en"
    HI = "hi"
    KN = "kn"
    TA = "ta"
    TE = "te"
    MR = "mr"
    BN = "bn"
    ML = "ml"
    GU = "gu"
    PA = "pa"


class Script(str, Enum):
    LATIN = "latin"
    DEVANAGARI = "devanagari"
    KANNADA = "kannada"
    TAMIL = "tamil"
    TELUGU = "telugu"
    MALAYALAM = "malayalam"
    GURMUKHI = "gurmukhi"
    GUJARATI = "gujarati"
    BENGALI = "bengali"


class LanguageConfig(BaseModel):
    model_config = ConfigDict(extra="ignore")

    code: LanguageCode
    name: str
    proficiency: Literal["primary", "secondary"]
    script: Script | None = None
